### DISCUSSION TITLE
Setting a new key in request object in nextjs middleware

### DISCUSSION THREAD

**Discussion by summit070** [vercel/next.js#70895] [App Router]
`// middleware.js    
export function middleware(request: NextRequest) {
    req.name = "foo";
    return NextResponse.next();
}`

`export default async function handler(req, res) {
    console.log( req.name ) //logs undefined
    // what can I do so that I can access the new properties
    // I put in the req object in the middlware file here?
}`

So, as you can see above, req.name is undefined in the route handler. How do i solve this issue

**Comment by icyJoseph** [comment]
Set it as a cookie:

> For incoming requests, cookies comes with the following methods: get, getAll, set, and delete cookies. You can check for the existence of a cookie with has or remove all cookies with clear.

IIRC, as long as it is not set on the outgoing response, the client won't see it.

Middleware and handlers, do not share the same JavaScript, runtime. Also your `function handler` looks more like an API route (from Pages Router), rather than a Router handler (from App Router)
